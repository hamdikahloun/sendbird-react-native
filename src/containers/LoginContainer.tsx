import React from 'react';
import {Image, Keyboard, StyleSheet, Text, View} from 'react-native';
import {Assets} from '@/assets';
import {LoginButton} from '@/components/loginButton';
import {NicknameInput} from '@/components/nicknameInput';
import {useUser} from '@/hooks';

export const LoginContainer = () => {
  const {loginUser, userSatus} = useUser();
  const [nickname, setNickname] = React.useState('');
  const loading = userSatus === 'loading';
  const disabled = !nickname || loading;

  const login = () => {
    Keyboard.dismiss();
    loginUser(nickname);
  };
  return (
    <View style={styles.container}>
      <Image style={styles.logo} source={Assets.logo} />
      <Text style={styles.title}>Hello,</Text>
      <Text style={styles.subtitle}>Please enter a nickname</Text>
      <NicknameInput onChangeNickname={setNickname} />
      <LoginButton onPress={login} disabled={disabled} loading={loading} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#F5F5F5',
    padding: 16,
  },

  logo: {
    resizeMode: 'center',
    alignSelf: 'center',
  },
  title: {
    color: '#817B8B',
    fontSize: 32,
  },
  subtitle: {
    color: '#0C0020',
    fontSize: 32,
  },
});
