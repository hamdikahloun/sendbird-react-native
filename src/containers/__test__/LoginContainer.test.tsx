import {renderWithReduxProviders} from '@/../testing';
import React from 'react';
import {LoginContainer} from '../LoginContainer';

describe('LoginContainer', () => {
  it('should render correctly', async () => {
    const screen = renderWithReduxProviders(<LoginContainer />).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
