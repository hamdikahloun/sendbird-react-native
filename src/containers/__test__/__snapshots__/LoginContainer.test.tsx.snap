// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/logo/logo.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "resizeMode": "center",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#817B8B",
        "fontSize": 32,
      }
    }
  >
    Hello,
  </Text>
  <Text
    style={
      Object {
        "color": "#0C0020",
        "fontSize": 32,
      }
    }
  >
    Please enter a nickname
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Nickname"
    style={
      Object {
        "backgroundColor": "#EFEFEF",
        "borderColor": "#F9F9F9",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 50,
        "marginBottom": 24,
        "marginTop": 8,
        "padding": 10,
      }
    }
    testID="nickname"
    value=""
  />
  <BVLinearGradient
    colors={
      Array [
        "#6100FF",
        "#8940FF",
      ]
    }
    style={
      Object {
        "borderRadius": 10,
        "height": 50,
        "marginHorizontal": 16,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="login-button"
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
        testID="login-button-text"
      >
        Login
      </Text>
    </View>
  </BVLinearGradient>
</View>
`;
